Description: Select correct DBus implementation for Banshee plugin harder.
Author: Christopher James Halse Rogers <raof@ubuntu.com>
Upstream: yes

Index: gnome-do-plugins/configure.ac
===================================================================
--- gnome-do-plugins.orig/configure.ac	2011-08-29 10:35:02.005394120 +1000
+++ gnome-do-plugins/configure.ac	2011-08-29 10:40:21.245397488 +1000
@@ -17,10 +17,7 @@
 AC_SUBST(ASM_VERSION)
 
 dnl pkg-config
-AC_PATH_PROG(PKG_CONFIG, pkg-config, no)
-if test "x$PKG_CONFIG" = "xno"; then
-        AC_MSG_ERROR([You need to install pkg-config])
-fi
+PKG_PROG_PKG_CONFIG
 
 AC_PROG_INSTALL
 
@@ -35,7 +32,25 @@
 
 dnl -- Banshee.CollectionIndexer.dll links to DBus# after 2.1, so the Banshee
 dnl -- plugin must follow suit.
-PKG_CHECK_EXISTS([banshee-collection-indexer >= 2.1], BANSHEE_USE_DBUS_SHARP=yes, BANSHEE_USE_DBUS_SHARP=no)
+AC_MSG_CHECKING([Banshee.CollectionIndexer package])
+PKG_CHECK_EXISTS([banshee-collection-indexer],
+		 [BANSHEE_COLLECTION_INDEXER=banshee-collection-indexer],
+		 [BANSHEE_COLLECTION_INDEXER=no])
+if test "x$BANSHEE_COLLECTION_INDEXER" = "xno" ; then
+	PKG_CHECK_EXISTS([banshee-collection-indexer-1],
+			 [BANSHEE_COLLECTION_INDEXER=banshee-collection-indexer-1],
+			 [BANSHEE_COLLECTION_INDEXER=not found])
+fi
+AC_MSG_RESULT([$BANSHEE_COLLECTION_INDEXER])
+
+AC_MSG_CHECKING([which DBus implementation Banshee.CollectionIndexer requires])
+if $PKG_CONFIG --print-requires $BANSHEE_COLLECTION_INDEXER 2>&1 | grep dbus-sharp-1.0 > /dev/null ; then
+	BANSHEE_USE_DBUS_SHARP=yes
+	AC_MSG_RESULT([dbus-sharp])
+else
+	BANSHEE_USE_DBUS_SHARP=no
+	AC_MSG_RESULT([ndesk-dbus])
+fi
 AM_CONDITIONAL(BANSHEE_DBUS_SHARP, test "x$BANSHEE_USE_DBUS_SHARP" = "xyes")
 
 dnl -- DBus# required for Banshee and Telepathy support
